packages/data-provider/src/services/submittedDataService.ts

```diff
@@ -195,25 +195,35 @@ const service = (dependencies: BaseDependencies) => {
 			// get dictionary relations
 			const dictionaryRelations = getDictionarySchemaRelations(dictionary);
 
-			const recordDependency = await searchDirectDependents(dictionaryRelations, submittedData);
+			const recordDependents = await searchDirectDependents(dictionaryRelations, foundRecordToDelete);
 
-			if (recordDependency && recordDependency.length > 0) {
-				recordsToUpdate.push(...recordDependency);
-			}
+			const submittedDataToDelete = [foundRecordToDelete, ...recordDependents];
 
-			if (dryRun === false) {
-				recordsToUpdate.forEach((record) => {
-					submittedDataRepo.delete(record, comment, userName);
-				});
-			}
+			const recordsToDeleteMap = transformmSubmittedDataToSubmissionDeleteData(submittedDataToDelete);
 
-			logger.info(
-				LOG_MODULE,
-				`Dry-Run '${dryRun}'`,
-				`Delete Submitted Data. Total records '${recordsToUpdate.length}'`,
-			);
+			// Get Active Submission or Open a new one
+			const activeSubmission = await getOrCreateActiveSubmission({
+				categoryId: foundRecordToDelete.dictionaryCategoryId,
+				userName,
+				organization: foundRecordToDelete.organization,
+			});
 
-			return mapRecordsSubmittedDataResponse(recordsToUpdate);
+			// Merge current Active Submission delete entities
+			const mergedSubmissionDeletes = mergeRecords(activeSubmission.data.deletes, recordsToDeleteMap);
```

Comment:
Just add items marked for deletion on the Submission instead of delete them right away